services/api/src/routes/users.js

```diff
@@ -113,19 +98,15 @@ router
       if (role) {
         query.roles = { $in: [role] };
       }
-      const data = await User.find(query)
-        .sort({ [sort.field]: sort.order === 'desc' ? -1 : 1 })
-        .skip(skip)
-        .limit(limit);
+      const { data, meta } = await search(User, query, body);
```

Comment:
there's some part of me really that wants to see these methods on the model... for one, having above `User.getSearchValidation` would parallel the work I'm doing with `model#getValidator` for create/patch Joi validations. This is maybe partially just an opinion, but there are a couple advantages to putting them on the model:

1. you wouldn't have to import anything... all models would have the various `getValidators` methods on them and a `search` method without importing in each route
2. models that require custom search behavior could further integrate it with the same API... for example:

```js
schema.methods.getSearchQuery = function getSearchQuery() {
  const query = super.getSearchQuery(); // I realize "super" doesn't make sense here but inheriting is kind of the intent
  // For various reasons we might want to exclude certain documents
  // or make other modifications to the query for this model
  query.someSpecialField = false;
  return query;
}
```

Reply:
The issue I have is that it ties this simple standalone utility further into the ORM. Perhaps we can leave this for later?

Reply:
I'll leave it to you... we can always change it later. But as far as I can tell it's already tied to the ORM, for example `search` isn't prepared to handle anything other than a mongoose model and it's calling methods like `exec` and `countDocuments`... also importing `ObjectId` directly from mongoose.

I get in a vague sense why you want to decouple things from the ORM but I think that should be driven by a specific scenario/use-case, otherwise I'm not really clear on what we're gaining here.

Also even if we have a specific need to decouple our models from the ORM there's no reason we can't do that at a higher level, as I see our models as "business logic" with the ORM layer only one aspect of them. Very simplistic example of what this could potentially look like if we want max-decoupling:

```js
// "Model" could be a class with methods shared across models but not
// specific to ORM... or maybe this isn't needed and just a basic class would be enough
class User extends Model {
  // User specific methods, props, etc
}

// Maybe some models do not make use of the ORM layer for some reason?
// If not this could be excluded.
mixinMongooseMethods(User);

// ES related facilities for models that use them? etc etc...
mixinSomethingElse(User);
```